apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_received += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+	return 0;
+}
+
+static int _udp_client(int size)
+{
+	struct sockaddr_in saddr;
+	int ret = 0;
+	int sockfd = 0;
+	
+	/* Socket Creation */
+	CT_LOG(TAG, "create socket");
+	sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		CT_LOGE(TAG, "create socket fail %d", errno);
+		return -1;
+	}
+	
+	/* Set Timeout for Sockets data transfer */
+	struct timeval tv;
+	tv.tv_sec = 30;
+	tv.tv_usec = 0;
+	ret = setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
+
+	/* Connect the socket to the server */
+	CT_LOG(TAG, "connect to %s:%d", WT_UDP_SERVER_ADDR, WT_UDP_PORT);
+	saddr.sin_family = AF_INET;
+	saddr.sin_port = htons(WT_UDP_PORT);
+	inet_aton(WT_UDP_SERVER_ADDR, &(saddr.sin_addr));
+	
+	unsigned int clen = sizeof(struct sockaddr_in);
+	int slen = sizeof(struct sockaddr_in);
+	CT_LOG(TAG, "Sending Device MAC to peer, mac: %s", g_mac_buf);
+	int nbytes = sendto(sockfd, g_mac_buf, sizeof(g_mac_buf), 0, (struct sockaddr*)&saddr, slen);
+	if (nbytes == 0) {
+		CT_LOGE(TAG, "connection closed");
+		return -1;
+	} else if (nbytes < 0) {
+		if (errno == EWOULDBLOCK) {			 
+			CT_LOGE(TAG, "timeout error %d", errno);
+			return -1;
+		} else {
+			CT_LOGE(TAG, "connection error %d", errno);
+			return -1;
+		}
+	}
+
+	CT_LOG(TAG, "Receiving MAC from peer");
+	nbytes = recvfrom(sockfd, g_recv_mac_buf, sizeof(g_recv_mac_buf), 0, (struct sockaddr*)&saddr, &clen);
+	if (nbytes == 0) {
+		CT_LOGE(TAG, "connection closed");
+		return -1;
+	} else if (nbytes < 0) {
+		if (errno == EWOULDBLOCK) {
+			CT_LOGE(TAG, "timeout error %d", errno);
+			return -1;
+		} else {
+			CT_LOGE(TAG, "connection error %d", errno);
+			return -1;
+		}
+	}
+	CT_LOG(TAG, "Peer MAC is: %s", g_recv_mac_buf);
+
+	/* Send data through udp socket */
+	ret = _udp_send_data(sockfd, size, &saddr);
+	if (ret < 0) {
+		CT_LOGE(TAG, "send fail size %d ret %d code %d", size, ret, errno);
+		close(sockfd);
+		return -1;
+	}
+	CT_LOG(TAG, "_udp_send_data OK");
+
+	CT_LOG(TAG, "receive ACK data size %d", WT_ACK_SIZE);
+	ret = _udp_recv_data(sockfd, WT_ACK_SIZE, &saddr);
+	if (ret < 0) {
+		CT_LOGE(TAG, "recv fail size %d ret %d code %d", WT_ACK_SIZE,ret, errno);
+		return -1;
+	}
+
+	/* Receive data through udp socket */
+	ret = _udp_recv_data(sockfd, size, &saddr);
+	if (ret < 0) {
+		CT_LOGE(TAG, "receive fail size %d ret %d code %d", size, ret, errno);
+		close(sockfd);
+		return -1;
+	}
+	CT_LOG(TAG, "_udp_recv_data OK");
+	
+	CT_LOG(TAG, "send ACK data size %d", WT_ACK_SIZE);
+	ret = _udp_send_data(sockfd, WT_ACK_SIZE, &saddr);
+	if (ret < 0) {
+		CT_LOGE(TAG, "send fail size %d ret %d code %d", WT_ACK_SIZE, ret, errno);
+		return -1;
+	}
+
+	close(sockfd);
+	CT_LOG(TAG, "udp data transmission finished");
+	return 0;
+}
+
+static int _tcp_send_data(int fd, int size)
```

Comment:
how about to implement udp and tcp seperate files. Then it will be easy to manage the code.

Reply:
Changes done as suggested